libcodechecker/libauth/cc_ldap.py

```diff
@@ -307,17 +341,36 @@ def auth_user(ldap_config, username=None, credentials=None):
         if group_base is None:
             LOG.error('Group base needs to be configured to'
                       'query ldap groups.')
-            return False
+            return []
 
-        member_pattern = ldap_config.get('groupMemberPattern')
-        if member_pattern is None or member_pattern == '':
-            member_pattern = '(member=$USERDN$)'
+        group_name_attr = ldap_config.get('groupNameAttr')
+        if group_name_attr is None:
+            LOG.error('groupNameAttr needs to be configured to'
+                      'query ldap groups.'
+                      'Its value must be the name'
+                      'attribute of the group.')
+            return []
 
-        member_pattern = member_pattern.replace('$USERDN$', user_dn)
-        member_query = '(& ' + group_pattern + member_pattern + ')'
+        # Attribute name must be ascii encoded
+        group_name_attr = group_name_attr.encode('ascii',
+                                                 'ignore')
+        attr_list = []
+        attr_list.append(group_name_attr)
 
-        is_member = check_group_membership(connection,
-                                           group_base,
-                                           member_query,
-                                           group_scope)
-        return is_member
+        LOG.debug("Performing LDAP search for group:" + group_pattern +
+                  "Group Name Attr:" + group_name_attr)
+
+        with ldap_error_handler():
+            group_result = connection.search_s(group_base,
+                                               group_scope,
+                                               group_pattern,
+                                               attr_list)
+        groups = []
+        for g in group_result:
+            groups.append(g[1][group_name_attr][0])
```

Comment:
Could you add some example what is the expected group format which is processed here?